docs/local-development.md

```diff
@@ -0,0 +1,143 @@
+# Local Development Setup
+
+## Prerequisites
+
+- Python 3.11+ (project requires >= 3.11)
+- Node.js 22+
+- npm
+
+## Installation
+
+### 1. Clone Repository
+
+```bash
+git clone https://github.com/mcp-getgather/mcp-getgather.git
+cd mcp-getgather
+```
+
+### 2. Install Dependencies
+
+#### Python Dependencies
+
+```bash
+# Using uv
+uv sync
+```
+
+#### Node.js Dependencies
+
+```bash
+npm install
+```
+
+## Running the Application
+
+### Development Mode
+
+Run both frontend and backend:
+
+```bash
+npm run dev
+```
+
+**Open in browser: `http://localhost:5173`**
+
+### Separate Development Servers
+
+#### Backend Only
+
+```bash
+npm run dev:backend
+# or
+uvicorn getgather.api.main:app --reload --host 127.0.0.1 --port 8000
+```
+
+**Open in browser: `http://localhost:8000`**
+
+#### Frontend Only
+
+```bash
+npm run dev:frontend
+# or
+vite
+```
+
+**Open in browser: `http://localhost:5173`**
```

Comment:
Not now but at some point we should make it so that opening 8000 *during* the development will give out the warning that this is not the right place to look at.